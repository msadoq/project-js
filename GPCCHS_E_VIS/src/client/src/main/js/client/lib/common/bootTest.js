import _ from 'lodash';
import { difference, intersection } from 'lodash/fp';
import { v4 } from 'uuid';
import path from 'path';

/*
  Setting up the testing framework before each test.
  Please see : https://facebook.github.io/jest/docs/cli.html#setuptestframeworkscriptfile-file
*/

global.testConfig = {
  ISIS_DOCUMENTS_ROOT: path.resolve(__dirname, '../documentManager/fixtures'),
  WILDCARD_CHARACTER: '*',
  VISUWINDOW_MAX_LENGTH: 42,
  STATE_COLORS: {
    alarm: 'orangered',
    critical: 'red',
    info: 'white',
    outOfRange: 'grey',
    severe: 'darkred',
    warning: 'orange',
    nonsignificant: 'lightgrey',
    obsolete: 'tan',
  },
  DEFAULT_FIELD: {},
};

_.set(
  global,
  'parameters.get',
  p => _.get(global.testConfig, p)
);

// jest expect.extend utils
const stringify = val => JSON.stringify(val, null, 2);

const toBe = (predicat, getAssertionString = _.identity) => (received, argument) => {
  const pass = predicat(received, argument);
  const assertionString = getAssertionString(stringify(argument));
  return {
    message: () => `expected ${stringify(received)}${pass ? ' not ' : ' '}to be ${assertionString}`,
    pass,
  };
};

function toHaveKeys(val, argument = []) {
  const keys = Object.keys(val);
  const diffKeys = (this.isNot ? intersection : difference)(argument, keys);
  const pass = this.isNot ? !!diffKeys.length : !diffKeys.length;
  const assertedKeys = JSON.stringify(diffKeys);
  return {
    message: () => `expected ${stringify(val)} to have${pass ? ' not ' : ' '}${assertedKeys} keys`,
    pass,
  };
}

const v4Length = v4().length;

// jest extended assertions
const extendedAssertions = {
  toBeArray: toBe(Array.isArray, () => 'an array'),
  toBeObject: toBe(x => typeof x === 'object', () => 'an object'),
  toBeString: toBe(x => typeof x === 'string', () => 'a string'),
  toBeOneOf: toBe((val, arg = []) => arg.includes(val), arg => `one of ${arg}`),
  toBeV4: toBe(x => typeof x === 'string' && x.length === v4Length, () => 'a V4 uuid'),
  toHaveKeys,
};

const aliases = {
  toBeAnArray: extendedAssertions.toBeArray,
  toBeAnObject: extendedAssertions.toBeObject,
  toBeAString: extendedAssertions.toBeString,
  toBeAnUuid: extendedAssertions.toBeV4,
};

expect.extend({
  ...extendedAssertions,
  ...aliases,
});
