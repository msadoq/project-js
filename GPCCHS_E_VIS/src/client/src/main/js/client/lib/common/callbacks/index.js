// ====================================================================
// HISTORY
// VERSION : 1.1.2 : DM : #5828 : 14/06/2017 : Cleanup client/ file organization and test helper modules
// END-HISTORY
// ====================================================================

import _isString from 'lodash/isString';
import _isEmpty from 'lodash/isEmpty';
import _has from 'lodash/has';
import _isFunction from 'lodash/isFunction';
import getLogger from '../../common/logManager';

const logger = getLogger('common:registeredCallbacks');

let callbacks = {};

const set = (id, callback) => {
  if (!_isString(id) || _isEmpty(id)) {
    throw new Error(`setting a new callback required a string id '${id}'`);
  }
  if (_has(callbacks, id)) {
    throw new Error(`a callback is already registered for this id '${id}'`);
  }
  if (!_isFunction(callback)) {
    throw new Error(`setting a new callback require a valid function '${id}'`);
  }

  logger.silly(`callback registered for '${id}'`);
  callbacks[id] = callback;
};

const get = id => callbacks[id];

const getAll = () => callbacks;

const remove = (id) => {
  delete callbacks[id];
};

const pop = (id) => {
  const callback = get(id);
  remove(id);
  return callback;
};

const clear = () => {
  logger.debug('callbacks cleared');
  callbacks = {};
};

module.exports = {
  set,
  get,
  pop,
  getAll,
  remove,
  clear,
};
