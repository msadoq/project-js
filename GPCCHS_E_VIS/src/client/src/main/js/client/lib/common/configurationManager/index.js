// ====================================================================
// HISTORY
// VERSION : 1.1.2 : DM : #5828 : 14/06/2017 : Move common/log and common/parameters in client/
// VERSION : 1.1.2 : FA : #6798 : 15/06/2017 : Add types.proto in dc - Add parse/stringify
//  mechanism to configurationManager
// VERSION : 1.1.2 : DM : #6700 : 21/06/2017 : Fix a configurationManager bug that ignore CLI
//  parameters.
// VERSION : 1.1.2 : FA : #6798 : 21/06/2017 : Modify configuration parameter to parse conf in
//  forked process
// VERSION : 1.1.2 : DM : #6700 : 21/06/2017 : Merge branch 'dev' into dbrugne-lifecycle
// VERSION : 1.1.2 : DM : #6700 : 21/06/2017 : Fix event loop blocking due to systematic JSON.parse
//  in configurationManager
// VERSION : 1.1.2 : DM : #6700 : 21/06/2017 : Fix important performance leak due to systematic
//  JSON.parse
// VERSION : 1.1.2 : FA : #6798 : 23/06/2017 : Merge branch 'dev' into pgaucher-464-proto-config
// VERSION : 1.1.2 : FA : #6798 : 27/06/2017 : branch 'dev' into pgaucher-464-proto-config
// VERSION : 1.1.2 : DM : #6700 : 06/07/2017 : Modify health monitoring mechanism : - Handle
//  properly start and stop - Add critical delay value in conf - Only start monitoring on DEBUG
// VERSION : 2.0.0 : FA : #8557 : 18/10/2017 : Add configuration file management for missions
//  adapters
// VERSION : 2.0.0 : FA : #8557 : 18/12/2017 : Fix load configuration file . .
// VERSION : 2.0.0 : FA : #10835 : 28/02/2018 : add global configuration for colors
// END-HISTORY
// ====================================================================

const fs = require('fs');
const { join } = require('path');
const minimist = require('minimist');
const _get = require('lodash/get');
const _merge = require('lodash/merge');
const _omit = require('lodash/omit');

const REQUIRED = 'config.required.json'; // required
const DEFAULT = 'config.default.json'; // mandatory
const LOCAL = 'config.local.json'; // optional

let fmdConfig = {};
let colorsConfig;
let argv;
let defaultConfig;
let localConfig;
let appRoot;

function setFmdConfiguration(config) {
  fmdConfig = config;
}

function getFmdConfiguration(name) {
  return fmdConfig[name];
}

function setColorsConfiguration(config) {
  colorsConfig = config;
}

function getColorsConfiguration(name) {
  if (!colorsConfig) {
    return undefined;
  }
  return colorsConfig[name];
}

function getAllArgv() {
  if (!argv) {
    argv = minimist(process.argv);
  }
  return argv;
}

function getArgv(name) {
  return getAllArgv()[name];
}

// TODO : remove this workaround in configurationManager refactoring
const env = process.env.mainProcessConfig
  ? JSON.parse(process.env.mainProcessConfig)
  : process.env;
function getEnv(name) {
  return env[name];
}

function getAllEnv() {
  return process.env;
}

function getLocal(name) {
  if (!localConfig) {
    return undefined;
  }
  return localConfig[name];
}

function getDefault(name) {
  if (!defaultConfig) {
    return undefined;
  }

  return defaultConfig[name];
}

/**
 * Search for argument name in:
 * - process.argv (cli)
 * - local values from file
 * - process.env (environment)
 * - default values from file
 */
function get(name) {
  let value;
  const hasValue = () => typeof value !== 'undefined';

  value = getFmdConfiguration(name);
  if (hasValue()) {
    return value;
  }
  value = getColorsConfiguration(name);
  if (hasValue()) {
    return value;
  }
  value = getArgv(name);
  if (hasValue()) {
    return value;
  }
  value = getLocal(name);
  if (hasValue()) {
    return value;
  }
  value = getEnv(name);
  if (hasValue()) {
    return value;
  }
  value = getDefault(name);
  if (hasValue()) {
    return value;
  }

  // return appRoot as path if not already found in previous configs
  if (name === 'path') {
    return appRoot;
  }

  return undefined;
}

function getAll() {
  const all = _merge(
    defaultConfig,
    getAllEnv(),
    localConfig,
    getAllArgv(),
    fmdConfig,
    colorsConfig
  );
  return _omit(all, ['_', 'r']);
}

function checkRequired(names) {
  names.forEach((name) => {
    const value = get(name);
    if (typeof value !== 'undefined') {
      return undefined;
    }

    throw new Error(`A required config parameter is missing: ${name}`);
  });
}

function init(path, checkForRequired = false) {
  // persist app root dir as 'path'
  appRoot = path;

  if (defaultConfig) {
    // was already inited
    return;
  }

  if (!fs.existsSync(join(path, DEFAULT))) {
    return;
  }

  defaultConfig = JSON.parse(fs.readFileSync(join(path, DEFAULT), 'utf8'));

  // optional local config file
  try {
    const localPath = join(path, LOCAL);
    fs.accessSync(localPath, fs.constants.F_OK);
    fs.accessSync(localPath, fs.constants.R_OK);
    localConfig = JSON.parse(fs.readFileSync(localPath, 'utf8'));
  } catch (e) {
    localConfig = {};
  }

  // check for required parameters
  if (!checkForRequired) {
    return;
  }
  const requiredConfig = JSON.parse(fs.readFileSync(join(path, REQUIRED), 'utf8')) || [];
  checkRequired(requiredConfig);
}

function universalGet(name) {
  return _get(global, ['parameters', 'get'], get)(name);
}

module.exports = { init, get: universalGet, getAll, setFmdConfiguration, setColorsConfiguration };
