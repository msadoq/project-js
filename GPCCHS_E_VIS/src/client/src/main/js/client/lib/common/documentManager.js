import createDocumentManager from '../documentManager';
import fmdApi from './fmd';

/**
 * documentsManager API
   please see documentation of this functions in lib/documentsManager/*
  {
    readViews:
    extractViews:
    saveView,
    saveViewAs,

    readPages,
    savePage,

    readWorkspace,
    saveWorkspace,
  }
*/

const documentManagerApi = createDocumentManager(fmdApi);
module.exports = documentManagerApi;
