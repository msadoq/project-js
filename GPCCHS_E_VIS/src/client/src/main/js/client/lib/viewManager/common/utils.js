import _ from 'lodash/fp';
import { buildFormulaForAutocomplete } from './index';

/**
 * @param data
 * @param id
 * @param defaultTimelineId
 * @returns {{name: *, connectedData: {formula: string, fieldX: string, timeline: *}}}
 * @pure
 */
export const parseDragData = (data, id, defaultTimelineId) => {
  const getComObject = _.propOr('UNKNOWN_COM_OBJECT', 0);
  const formula =
    buildFormulaForAutocomplete(
      data.catalogName,
      data.item,
      getComObject(data.comObjects),
      data.comObjectFields
    );
  return {
    name: data.item,
    connectedData: {
      formula,
      catalog: data.catalogName,
      catalogItem: data.item,
      comObject: getComObject(data.comObjects),
      comObjectField: data.comObjectFields || 'convertedValue',
      fieldX: 'onboardDate',
      unit: 'V',
      domain: data.domain || '*',
      timeline: defaultTimelineId,
    },
  };
};

export const dummy = (x, y) => x + y;
