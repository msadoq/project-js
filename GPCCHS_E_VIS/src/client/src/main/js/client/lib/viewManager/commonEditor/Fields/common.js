import { get } from 'common/configurationManager';

/**
 * @param list
 * @param addWildcard
 */
// eslint-disable-next-line import/prefer-default-export
export const computeOptions = (list, addWildcard = false) => {
  let options =
    list && Array.isArray(list)
      ? list.map(d =>
        ({
          label: d.name,
          value: d.name,
        })
      )
      : []
  ;
  if (addWildcard) {
    const wildcardCharacter = get('WILDCARD_CHARACTER');
    if (!options.find(s => s.value === wildcardCharacter)) {
      options = options.concat({
        label: wildcardCharacter,
        value: wildcardCharacter,
      });
    }
  }

  return options;
};
