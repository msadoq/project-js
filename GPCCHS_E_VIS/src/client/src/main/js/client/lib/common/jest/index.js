// ====================================================================
// HISTORY
// VERSION : 1.1.2 : FA : #6670 : 16/06/2017 : Move and rename jest.js in jest/setup.js + test.js in jest/index.js
// VERSION : 1.1.2 : FA : #6798 : 22/06/2017 : Multiple changes on the load mechansim of adapters : - To test with Jest, add a mock of config(MESSAGES_NAMESPACE) in jest/index.js - Test fix - Lint pass ( 1 test is still KO)
// VERSION : 1.1.2 : DM : #6700 : 28/06/2017 : Automatically freeze given state to mockStore (in common/jest)
// VERSION : 1.1.2 : FA : #7453 : 07/08/2017 : Change adapters path in tests
// END-HISTORY
// ====================================================================

import { tmpdir } from 'os';
import _ from 'lodash/fp';
import { resolve } from 'path';
import thunk from 'redux-thunk';
import deepFreeze from 'deep-freeze';
import configureMockStore from 'redux-mock-store';
import { loadStubs, getStubData } from '../../utils/stubs';
import flattenDataId from '../flattenDataId';
import { registerGlobal } from '../../utils/adapters';

const mockedAdaptersPath = [
  {
    path: './adapters/',
    ns: 'isis',
  },
  {
    path: './adapters/',
    ns: 'dc',
  },
];

const mockLoadStubs = () => loadStubs(mockedAdaptersPath);
const mockRegister = () => registerGlobal(mockedAdaptersPath);


const dataStub = getStubData();

const freezeMe = o => o && deepFreeze(o);

const freezeArgs = f => (...args) => {
  const frozenArgs = args.map(arg => freezeMe(arg));
  return f(...frozenArgs);
};

const mockStore = _.compose(configureMockStore([thunk]), freezeMe);

const testMemoization = (selector, state, ownProps) => {
  const newState = _.cloneDeep(state);
  const newOwnProps = _.cloneDeep(ownProps);
  selector.resetRecomputations();
  expect(selector.recomputations()).toBe(0);

  const result1 = selector(newState, newOwnProps);
  expect(selector.recomputations()).toBe(1);

  const result2 = selector(newState, newOwnProps);
  expect(selector.recomputations()).toBe(1);
  expect(result1).toEqual(result2);
};

const getRemoteId = override => flattenDataId(dataStub.getDataId(override));

const mockIpcReply = () => {
  //
};

module.exports = {
  mockStore, // thunk testing
  mockIpcReply, // IPC controller testing
  freezeMe, // reducers testing
  freezeArgs, // reducers testing
  testMemoization, // reselect testing
  mockRegister, // protobuf testing
  mockLoadStubs,
  getTmpPath: (...args) => resolve(tmpdir(), 'vima-tests', ...args), // documentManager testing
  getRemoteId,
};
