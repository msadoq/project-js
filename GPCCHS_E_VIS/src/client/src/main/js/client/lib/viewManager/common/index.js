import _unset from 'lodash/unset';
import _memoize from 'lodash/memoize';
import { SIGNIFICANT_VALIDITY_STATE_VALUE } from 'constants';
import { SDB_VALUE_OPTION, TIME_BASED_DATA_OPTION } from '../commonEditor/Fields/DataTypeField';

export const buildFormula = (catalog, catalogItem, comObject, comObjectField) => {
  let result = '';
  if (catalog) {
    result += catalog;
  }

  if (catalog && catalogItem) {
    result += `.${catalogItem}`;
  }

  if (catalog && catalogItem && comObject) {
    result += `<${comObject}>`;
  }

  if (catalog && catalogItem && comObject && comObjectField) {
    result += `.${comObjectField}`;
  }

  return result;
};

export const buildFormulaForAutocomplete = (catalog, catalogItem, comObject, comObjectField) => {
  let result = '';
  if (catalog) {
    result += catalog;

    if (catalogItem) {
      result += `.${catalogItem}`;

      if (comObject) {
        result += `<${comObject}>`;

        if (comObject.indexOf('ReportingParameter') > -1) {
          result += '.convertedValue';
          return result;
        } else if (comObject.indexOf('DecommutedPacket') > -1) {
          return result;
        }

        if (comObjectField) {
          result += `.${comObjectField}`;
        }

        return result;
      }
    }
  }
  return result;
};

/**
 * Handles entry point editor form submission
 * @param values
 * @param updateEntryPoint callback to update the entry point
 * @param viewId
 */
export function handleSubmit(values, updateEntryPoint, viewId) {
  const { dataType } = values.connectedData;
  if (dataType !== SDB_VALUE_OPTION.value) {
    _unset(values.connectedData, 'path');
    _unset(values.connectedData, 'displayMode');
  }
  if (dataType !== TIME_BASED_DATA_OPTION.value) {
    _unset(values.connectedData, 'catalogItem');
    _unset(values.connectedData, 'comObject');
    _unset(values.connectedData, 'comObjectField');
    _unset(values.connectedData, 'provider');
    _unset(values.connectedData, 'refTimestamp');
  }
  const { catalog, catalogItem, comObject, comObjectField } = values.connectedData;
  const formula = buildFormula(catalog, catalogItem, comObject, comObjectField);
  updateEntryPoint(viewId, values.id, {
    ...values,
    connectedData: {
      ...values.connectedData,
      formula,
    },
  });
}

/**
 * given an array of required fields, returns the validate function
 * @param requiredFields
 * @returns a redux-form validate function
 */
const innerValidateRequiredFields = requiredFields => (values = {}) => {
  const errors = {};

  requiredFields.forEach((field) => {
    if (!values[field]) {
      errors[field] = 'Required';
    }
  });
  return errors;
};

export const validateRequiredFields = (requiredFields, values) => (
  values
    ? innerValidateRequiredFields(requiredFields)(values) // executes the validation on values
    : innerValidateRequiredFields(requiredFields) // returns a validation function
);

/**
 * given a validity state, returns true if value is "VALID(2)" or false otherwise
 * @type {Function}
 * @returns boolean
 */
export const memoizeIsSignificantValue = _memoize(
  validityState => validityState === SIGNIFICANT_VALIDITY_STATE_VALUE
);
