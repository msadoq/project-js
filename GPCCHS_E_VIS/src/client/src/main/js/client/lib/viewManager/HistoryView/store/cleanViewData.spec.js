import _cloneDeep from 'lodash/cloneDeep';
import _omit from 'lodash/omit';
import cleanCurrentViewData, { updateEpLabel, removeViewDataByEp } from './cleanViewData';
import { freezeMe } from '../../../common/jest';
import state from '../../../common/jest/stateTest';
import dataMapGenerator from '../../../dataManager/map';


describe('viewManager/HistoryView/store/cleanViewData', () => {
  const dataMap = dataMapGenerator(state);
  const viewMap = dataMap.perView;
  const historyConfig = state.HistoryViewConfiguration;

  describe('cleanCurrentViewData', () => {
    test('no update', () => {
      const frozen = freezeMe(state.HistoryViewData.hist1);
      expect(
        cleanCurrentViewData(frozen, viewMap.hist1, viewMap.hist1, dataMap.expectedRangeIntervals,
          dataMap.expectedRangeIntervals, historyConfig)).toBe(frozen);
    });
    test('interval update History: keep all', () => {
      const newMap = _cloneDeep(viewMap);
      const newIntervals = _cloneDeep(dataMap.expectedRangeIntervals);
      newIntervals['Reporting.ATT_BC_REVTCOUNT1<ReportingParameter>:0:1']['groundDate/extractedValue.tb1:0']
      .expectedInterval[1] += 5000;
      const frozen = freezeMe(state.HistoryViewData.hist1);
      expect(
        cleanCurrentViewData(frozen, viewMap.hist1, newMap.hist1, dataMap.expectedRangeIntervals,
          newIntervals, historyConfig)).toBe(frozen);
    });
    test('interval update History: keep some', () => {
      const newMap = _cloneDeep(viewMap);
      const newIntervals = _cloneDeep(dataMap.expectedRangeIntervals);
      newIntervals['Reporting.ATT_BC_REVTCOUNT1<ReportingParameter>:0:1']['groundDate/extractedValue.tb1:0']
      .expectedInterval[1] += 5000;
      newIntervals['Reporting.ATT_BC_REVTCOUNT1<ReportingParameter>:0:1']['groundDate/extractedValue.tb1:0']
      .expectedInterval[0] += 5000;
      const newState = cleanCurrentViewData(freezeMe(state.HistoryViewData.hist1), viewMap.hist1,
        newMap.hist1, dataMap.expectedRangeIntervals, newIntervals, historyConfig);
      expect(newState).toMatchSnapshot();
    });
    test.skip('interval update History: remove all', () => {
      const newMap = _cloneDeep(viewMap);
      const newIntervals = _cloneDeep(dataMap.expectedRangeIntervals);
      newIntervals['Reporting.TMMGT_BC_VIRTCHAN3<ReportingParameter>:0:4:rawValue.>.100']['extractedValue.tb1:0']
        .expectedInterval = [500030, 900000];
      newIntervals['Reporting.ATT_BC_REVTCOUNT1<ReportingParameter>:0:1']['groundDate/extractedValue.tb1:0']
        .expectedInterval = [500030, 900000];
      const newState = cleanCurrentViewData(freezeMe(state.HistoryViewData.hist1), viewMap.hist1,
              newMap.hist1, dataMap.expectedRangeIntervals, newIntervals, historyConfig);
      expect(newState).toEqual(
        { cols: state.HistoryViewData.hist1.cols, lines: [], data: {}, indexes: {} });
    });
    test('interval error History: remove all', () => {
      const newMap = _cloneDeep(viewMap);
      const newState = cleanCurrentViewData(freezeMe(state.HistoryViewData.hist1), viewMap.hist1,
        newMap.hist1, dataMap.expectedRangeIntervals, undefined, historyConfig);
      expect(newState).toEqual(
        { cols: state.HistoryViewData.hist1.cols, lines: [], data: {}, indexes: {} });
    });
    test('Ep renaming', () => {
      const newMap = _cloneDeep(viewMap);
      newMap.hist1.entryPoints = {
        ...newMap.hist1.entryPoints,
        ATT_BC_REVTCOUNT10: Object.assign({}, newMap.hist1.entryPoints.ATT_BC_REVTCOUNT1),
      };
      newMap.hist1.entryPoints = _omit(newMap.hist1.entryPoints, 'ATT_BC_REVTCOUNT1');
      expect(cleanCurrentViewData(Object.freeze(state.HistoryViewData.hist1), viewMap.hist1,
        newMap.hist1, dataMap.expectedRangeIntervals, dataMap.expectedRangeIntervals
        , historyConfig))
        .toMatchSnapshot();
    });
  });
  describe('updateEpLabel', () => {
    test('values ok', () => {
      expect(updateEpLabel(freezeMe(state.HistoryViewData.hist1), 'ATT_BC_REVTCOUNT1', 'ATT_BC_REVTCOUNT10'))
      .toEqual({ cols:
      ['referenceTimestamp',
        'extractedValue',
        'rawValue',
        'monitoringState'],
        lines:
        ['ATT_BC_REVTCOUNT10 100020',
          'ATT_BC_REVTCOUNT10 200020',
          'ATT_BC_REVTCOUNT10 300020',
          'ATT_BC_REVTCOUNT10 400020',
          'ATT_BC_REVTCOUNT10 500020'],
        indexes: { ATT_BC_REVTCOUNT10: ['100020', '200020', '300020', '400020', '500020'] },
        data: { ATT_BC_REVTCOUNT10: state.HistoryViewData.hist1.data.ATT_BC_REVTCOUNT1 },
      });
    });
    test('unknown value', () => {
      const frozen = freezeMe(state.HistoryViewData.hist1);
      expect(updateEpLabel(frozen, 'ATT_BC_REVTCOUNT10', 'ATT_BC_REVTCOUNT100')).toBe(frozen);
    });
  });
  describe('removeViewDataByEp', () => {
    test('should support empty state', () => {
      const frozen = freezeMe({});
      expect(removeViewDataByEp(frozen, 'TMMGT_BC_VIRTCHAN3', 10, 20)).toBe(frozen);
      const otherFrozen = freezeMe({ cols: [], lines: [], data: {}, indexes: {} });
      expect(removeViewDataByEp(otherFrozen, 'TMMGT_BC_VIRTCHAN3', 10, 20)).toBe(otherFrozen);
    });
    test('should support nothing to keep', () => {
      expect(removeViewDataByEp(freezeMe(state.HistoryViewData.hist1), 'ATT_BC_REVTCOUNT1', 0, 100))
      .toMatchSnapshot();
    });
    test('upper < lower: nothing to keep', () => {
      expect(removeViewDataByEp(freezeMe(state.HistoryViewData.hist1), 'ATT_BC_REVTCOUNT1', 10000, 100))
      .toMatchSnapshot();
    });
    test('should support partial keeping', () => {
      expect(removeViewDataByEp(freezeMe(state.HistoryViewData.hist1), 'ATT_BC_REVTCOUNT1', 100000, 250000))
      .toMatchSnapshot();
    });
    test('should support keep everything', () => {
      expect(removeViewDataByEp(freezeMe(state.HistoryViewData.hist1), 'ATT_BC_REVTCOUNT1', 100000, 600000))
      .toMatchSnapshot();
    });
  });
});
