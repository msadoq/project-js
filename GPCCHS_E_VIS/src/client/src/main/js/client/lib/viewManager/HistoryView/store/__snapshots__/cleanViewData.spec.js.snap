// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewManager/HistoryView/store/cleanViewData cleanCurrentViewData Ep renaming 1`] = `
Object {
  "cols": Array [
    "referenceTimestamp",
    "extractedValue",
    "rawValue",
    "monitoringState",
  ],
  "data": Object {
    "ATT_BC_REVTCOUNT10": Object {
      "100020": Object {
        "extractedValue": 149.30028143545633,
        "masterTime": 100020,
        "monitoringState": "valid",
        "rawValue": 149.30028143545633,
        "referenceTimestamp": 100000,
      },
      "200020": Object {
        "extractedValue": 154.3002814355,
        "masterTime": 200020,
        "monitoringState": "valid",
        "rawValue": 154.3002814355,
        "referenceTimestamp": 200000,
      },
      "300020": Object {
        "extractedValue": 159.3002814355,
        "masterTime": 300020,
        "monitoringState": "invalid",
        "rawValue": 159.3002814355,
        "referenceTimestamp": 300000,
      },
      "400020": Object {
        "extractedValue": 164.3002814355,
        "masterTime": 400020,
        "monitoringState": "obsolete",
        "rawValue": 164.3002814355,
        "referenceTimestamp": 400000,
      },
      "500020": Object {
        "extractedValue": 169.3002814355,
        "masterTime": 500020,
        "monitoringState": "valid",
        "rawValue": 169.3002814355,
        "referenceTimestamp": 500000,
      },
    },
  },
  "indexes": Object {
    "ATT_BC_REVTCOUNT1": Array [],
  },
  "lines": Array [
    "ATT_BC_REVTCOUNT1 100020",
    "ATT_BC_REVTCOUNT1 200020",
    "ATT_BC_REVTCOUNT1 300020",
    "ATT_BC_REVTCOUNT1 400020",
    "ATT_BC_REVTCOUNT1 500020",
  ],
}
`;

exports[`viewManager/HistoryView/store/cleanViewData cleanCurrentViewData interval update History: keep some 1`] = `
Object {
  "cols": Array [
    "referenceTimestamp",
    "extractedValue",
    "rawValue",
    "monitoringState",
  ],
  "data": Object {
    "ATT_BC_REVTCOUNT1": Object {
      "100020": Object {
        "extractedValue": 149.30028143545633,
        "masterTime": 100020,
        "monitoringState": "valid",
        "rawValue": 149.30028143545633,
        "referenceTimestamp": 100000,
      },
      "200020": Object {
        "extractedValue": 154.3002814355,
        "masterTime": 200020,
        "monitoringState": "valid",
        "rawValue": 154.3002814355,
        "referenceTimestamp": 200000,
      },
      "300020": Object {
        "extractedValue": 159.3002814355,
        "masterTime": 300020,
        "monitoringState": "invalid",
        "rawValue": 159.3002814355,
        "referenceTimestamp": 300000,
      },
      "400020": Object {
        "extractedValue": 164.3002814355,
        "masterTime": 400020,
        "monitoringState": "obsolete",
        "rawValue": 164.3002814355,
        "referenceTimestamp": 400000,
      },
      "500020": Object {
        "extractedValue": 169.3002814355,
        "masterTime": 500020,
        "monitoringState": "valid",
        "rawValue": 169.3002814355,
        "referenceTimestamp": 500000,
      },
    },
  },
  "indexes": Object {
    "ATT_BC_REVTCOUNT1": Array [],
  },
  "lines": Array [
    "ATT_BC_REVTCOUNT1 100020",
    "ATT_BC_REVTCOUNT1 200020",
    "ATT_BC_REVTCOUNT1 300020",
    "ATT_BC_REVTCOUNT1 400020",
    "ATT_BC_REVTCOUNT1 500020",
  ],
}
`;

exports[`viewManager/HistoryView/store/cleanViewData removeViewDataByEp should support keep everything 1`] = `
Object {
  "cols": Array [
    "referenceTimestamp",
    "extractedValue",
    "rawValue",
    "monitoringState",
  ],
  "data": Object {
    "ATT_BC_REVTCOUNT1": Object {
      "100020": Object {
        "extractedValue": 149.30028143545633,
        "masterTime": 100020,
        "monitoringState": "valid",
        "rawValue": 149.30028143545633,
        "referenceTimestamp": 100000,
      },
      "200020": Object {
        "extractedValue": 154.3002814355,
        "masterTime": 200020,
        "monitoringState": "valid",
        "rawValue": 154.3002814355,
        "referenceTimestamp": 200000,
      },
      "300020": Object {
        "extractedValue": 159.3002814355,
        "masterTime": 300020,
        "monitoringState": "invalid",
        "rawValue": 159.3002814355,
        "referenceTimestamp": 300000,
      },
      "400020": Object {
        "extractedValue": 164.3002814355,
        "masterTime": 400020,
        "monitoringState": "obsolete",
        "rawValue": 164.3002814355,
        "referenceTimestamp": 400000,
      },
      "500020": Object {
        "extractedValue": 169.3002814355,
        "masterTime": 500020,
        "monitoringState": "valid",
        "rawValue": 169.3002814355,
        "referenceTimestamp": 500000,
      },
    },
  },
  "indexes": Object {
    "ATT_BC_REVTCOUNT1": Array [
      "100020",
      "200020",
      "300020",
      "400020",
      "500020",
    ],
  },
  "lines": Array [
    "ATT_BC_REVTCOUNT1 100020",
    "ATT_BC_REVTCOUNT1 200020",
    "ATT_BC_REVTCOUNT1 300020",
    "ATT_BC_REVTCOUNT1 400020",
    "ATT_BC_REVTCOUNT1 500020",
  ],
}
`;

exports[`viewManager/HistoryView/store/cleanViewData removeViewDataByEp should support nothing to keep 1`] = `
Object {
  "cols": Array [
    "referenceTimestamp",
    "extractedValue",
    "rawValue",
    "monitoringState",
  ],
  "data": Object {
    "ATT_BC_REVTCOUNT1": Object {},
  },
  "indexes": Object {
    "ATT_BC_REVTCOUNT1": Array [
      "100020",
      "200020",
      "300020",
      "400020",
      "500020",
    ],
  },
  "lines": Array [
    "ATT_BC_REVTCOUNT1 100020",
    "ATT_BC_REVTCOUNT1 200020",
    "ATT_BC_REVTCOUNT1 300020",
    "ATT_BC_REVTCOUNT1 400020",
    "ATT_BC_REVTCOUNT1 500020",
  ],
}
`;

exports[`viewManager/HistoryView/store/cleanViewData removeViewDataByEp should support partial keeping 1`] = `
Object {
  "cols": Array [
    "referenceTimestamp",
    "extractedValue",
    "rawValue",
    "monitoringState",
  ],
  "data": Object {
    "ATT_BC_REVTCOUNT1": Object {
      "100020": Object {
        "extractedValue": 149.30028143545633,
        "masterTime": 100020,
        "monitoringState": "valid",
        "rawValue": 149.30028143545633,
        "referenceTimestamp": 100000,
      },
      "200020": Object {
        "extractedValue": 154.3002814355,
        "masterTime": 200020,
        "monitoringState": "valid",
        "rawValue": 154.3002814355,
        "referenceTimestamp": 200000,
      },
    },
  },
  "indexes": Object {
    "ATT_BC_REVTCOUNT1": Array [
      "100020",
      "200020",
      "300020",
      "400020",
      "500020",
    ],
  },
  "lines": Array [
    "ATT_BC_REVTCOUNT1 100020",
    "ATT_BC_REVTCOUNT1 200020",
    "ATT_BC_REVTCOUNT1 300020",
    "ATT_BC_REVTCOUNT1 400020",
    "ATT_BC_REVTCOUNT1 500020",
  ],
}
`;

exports[`viewManager/HistoryView/store/cleanViewData removeViewDataByEp upper < lower: nothing to keep 1`] = `
Object {
  "cols": Array [
    "referenceTimestamp",
    "extractedValue",
    "rawValue",
    "monitoringState",
  ],
  "data": Object {
    "ATT_BC_REVTCOUNT1": Object {},
  },
  "indexes": Object {
    "ATT_BC_REVTCOUNT1": Array [
      "100020",
      "200020",
      "300020",
      "400020",
      "500020",
    ],
  },
  "lines": Array [
    "ATT_BC_REVTCOUNT1 100020",
    "ATT_BC_REVTCOUNT1 200020",
    "ATT_BC_REVTCOUNT1 300020",
    "ATT_BC_REVTCOUNT1 400020",
    "ATT_BC_REVTCOUNT1 500020",
  ],
}
`;
